WordPress/openverse-catalog
# Refactor Phylopic to use ProviderDataIngester

## Fixes


Fixes #591 by @stacimc

## Description


This PR refactors the Phylopic provider script to use the new `ProviderDataIngester` class. 

This refactor is actually pretty significant because the query params are **not** the defining feature for determining the values to retrieve, the `endpoint` is! I had to usurp existing functions a bit in order to get this all to work with the ingester class. Let me know if the current setup is too confusing or difficult to work with. Frankly we can strip out the logic for processing the entire dataset if we'd like and just make this a dated DAG. 

## Testing Instructions

1. `just recreate && just test`
1. Set the class's `delay` attribute to `0.5` and turn the DAG on. It will take about 5 runs (due to catchup), but the 5th run should actually capture data.
1. Run `just run python openverse_catalog/dags/providers/provider_api_scripts/phylopic.py --date-start 2020-11-15`, this should ingest 6 records
1. (Optionally) Run `just run python openverse_catalog/dags/providers/provider_api_scripts/phylopic.py`, this should attempt to ingest everything, but let it go for only a few iterations before stopping

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
